### PR TITLE
smoker trait ligther is a 'bonus' trinket spawn

### LINKED ISSUE #15389: Smoker trait overrides Trinket trait (Lunchbox)
### Labels

[BUG][MINOR]

### Description

Took smoker trait and lunchbox trait and the lighter it gave me overrode my lunchbox trait

### Steps to reproduce

-  Take smoker trait
-  Take Lunchbox trait
-  Lighter but no lunchbox!

### Expected Behavior

Either you get both a lunchbox and a lighter or the smoker trait should be mutually exclusive with trinket traits like they are with each other

### Additional Information & Screenshots

Reported by: Joeled
On server: Goonstation 4 Roleplay: Sylvester
Active test merges: nothing
Round log date: 2023-08-15-01-59
Reported on: 2023-08-15 02:05:52
Map: NADIR

### PR DESCRIPTION
**Pull request by pgmzeta**
[traits]
## About the PR 
Remove the smoker trait from trinket checks, and instead add a trinket-like zippo lighter as a second item.

## Why's this needed? 
Fixes #15389

### CODE REVIEW

**Comment by AXLFAN2000** [comment]
you are my hero forever

**Review by TealSeer — APPROVED** [review]

File: code/procs/jobprocs.dm [hunk 1]
```diff
@@ -676,39 +676,40 @@ var/global/totally_random_jobs = FALSE
 		qdel(pda)
 
 	var/T = pick(trinket_safelist)
-	var/obj/item/trinket = null
+	var/list/obj/item/trinkets = list()
 
 	if (src.traitHolder && src.traitHolder.hasTrait("pawnstar"))
```
Comment (ZeWaka) [inline code/procs/jobprocs.dm:681]: can you check that the traitholder exists at the top and remove all these dumb useless checks? really gross

**Review by ZeWaka — CHANGES_REQUESTED** [review]

File: code/procs/jobprocs.dm (hunk 1 above)
Comment (pgmzeta) [reply code/procs/jobprocs.dm]: [af4cff0](https://github.com/goonstation/goonstation/pull/16352/commits/af4cff0762dd37ec9f11090894ae849e3ed1a1b4)

**Review by ZeWaka — APPROVED** [review]

**Comment by Tarmunora** [comment]
how does this interact with spythief trinket targeting?

**Comment by pgmzeta** [comment]
> how does this interact with spythief trinket targeting?

A read of the code suggests incorrectly; moving to draft while I work out a solution

**Pull request merged**